website/src/config.ts

```diff
@@ -18,9 +18,33 @@ function getConfigDir(): string {
     return configDir;
 }
 
+function validateWebsiteConfig(config: WebsiteConfig): Error[] {
```

Comment:
I suspect think this is probably OK, but we should be wary that we need to get the config for pretty much every single thing we ever do on the website, so if we start to add significant milliseconds of validation that's probably a problem, so we should be wary of expanding this pattern too much

Reply:
The validated config is cached in _config, validation only happens on initial read

Reply:
```typescript
export function getWebsiteConfig(): WebsiteConfig {
    if (_config === null) {
        _config = readTypedConfigFile('website_config.json', websiteConfig);
        const config = readTypedConfigFile('website_config.json', websiteConfig);
        const validationErrors = validateWebsiteConfig(config);
        if (validationErrors.length > 0) {
            throw new AggregateError(validationErrors, 'There were validation errors in the website_config.json');
        }
        _config = config;
    }
    return _config;
}
```

But now that I read it, if there are errors, they will keep getting parsed 😅

Reply:
Are you saying you think this code won't be executed every time a page is loaded? I'm 90% sure that it will - you can check by adding a print statement.

Reply:
(but no that's probably not what you are saying - I was unclear, I meant "every route" essentially)

Reply:
I got curious! I just checked with a print. The part inside of the `if` is only executed once! My understanding is that `_config` is a global variable that exists once in the server, so it is also there across page loads.

Reply:
oh what! ok sorry, my bad!!

Reply:
Thanks for educating me :)